bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/ShellyHandlerFactory.java

```diff
@@ -130,6 +142,7 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
     @Override
     protected synchronized void removeHandler(@NonNull ThingHandler thingHandler) {
         if (thingHandler instanceof ShellyBaseHandler) {
+            ((ShellyBaseHandler) thingHandler).stop();
```

Comment:
Sometimes I see "xxx, but thing handled is already disposed". Maybe this is a timing issue when shutting down the thing, but another event comes in at that time. Put it in just to avoid this potential scenario, but I could also remove it

Reply:
removed it